src/main/java/com/marklogic/spark/reader/document/DocumentRowSchema.java

```diff
@@ -8,7 +8,15 @@ public abstract class DocumentRowSchema {
     public static final StructType SCHEMA = new StructType()
         .add("URI", DataTypes.StringType)
         .add("content", DataTypes.BinaryType)
-        .add("format", DataTypes.StringType);
+        .add("format", DataTypes.StringType)
+        .add("collections", DataTypes.createArrayType(DataTypes.StringType))
+        .add("permissions", DataTypes.createMapType(
```

Comment:
The Spark `MapType` and `ArrayType` allow us to model just about any data structure returned by the server.